data-prepper-plugins/cloudwatch-logs/src/main/java/org/opensearch/dataprepper/plugins/sink/client/CloudWatchLogsClientFactory.java

```diff
@@ -20,6 +20,9 @@
  * CloudWatchLogs services.
  */
 public final class CloudWatchLogsClientFactory {
+    private CloudWatchLogsClientFactory() {
+        throw new IllegalStateException("Static Factory Class!");
+    }
 
     /**
      * Generates a CloudWatchLogs Client based on STS role ARN system credentials.
```

Comment:
In the `createOverrideConfiguration()` method you use nested builder. It is better practice to use lambdas instead (easier to read).
```
return ClientOverrideConfiguration.builder()
            .retryPolicy(r -> r.numRetries(AwsConfig.DEFAULT_CONNECTION_ATTEMPTS))
            .build();
```

Reply:
Thank you! I have added the lambda to increase code readability.